src/chart/main.c

```diff
@@ -868,51 +868,77 @@ static char *encode_colorchecker(int num, const double *point, const double **ta
   return dt_exif_xmp_encode_internal((uint8_t *)&params, sizeof(params), NULL, FALSE);
 }
 
+static int compare_L_source(const void *x_, const void *y_) {
+  const double *x = x_;
+  const double *y = y_;
+  return (*x) < (*y) ? -1 : ((*x) > (*y) ? +1 : 0);
```

Comment:
```c++
static int compare_L_source(const void *x_, const void *y_) {
  const double x = *x_;
  const double y = *y_;
  return x < y ? -1 : (x > y ? +1 : 0);
}
```

Reply:
This implementation of the comparison function is more easy to read, indeed.

Reply:
... but this is not valid C. A type cast needs to be applied before dereferencing the pointers.

Reply:
Ah well :)
So either
```c++
static int compare_L_source(const void *x_, const void *y_) {
  const double x = *(const double*)x_;
  const double y = *(const double*)y_;
  return x < y ? -1 : (x > y ? +1 : 0);
}
```

or
```c++
static int compare_L_source(const double *x_, const double *y_) {
  const double x = *x_;
  const double y = *y_;
  return x < y ? -1 : (x > y ? +1 : 0);
}
...
  qsort(grays, num_tonecurve, sizeof(double) * 6, (int (*)(const void *, const void *))compare_L_source);
```

Reply:
Both variants will do the job. Personally I would prefer the 1st one. Function pointer casts are difficult to read.